packages/quantic/force-app/main/default/lwc/quanticCaseClassification/quanticCaseClassification.js

```diff
@@ -185,27 +185,17 @@ export default class QuanticCaseClassification extends LightningElement {
   }
 
   updateFieldState() {
-    if (this.maxSuggestions > 0 && !this.lockedState) {
-      this.previousClassifications = this.classifications;
-      this.classifications =
-        this.field.state.suggestions.slice(
-          0,
-          Math.max(Number(this.maxSuggestions), 0)
-        ) ?? [];
+    if (!this.lockedState) {
       this.loading = this.field.state.loading;
-      if (this.newSuggestionsReceived) {
+      if (this.maxSuggestions > 0 && this.newSuggestionsReceived) {
```

Comment:
You could simplify a bit I think?

```suggestion
const hasNewSuggestions = this.maxSuggestions > 0 && this.newSuggestionsReceived;
const value = (hasNewSuggestions && this.isAutoSelectionNeeded) ? this.classifications[0].value : this.field.state.value;
this.setFieldValue(value, hasNewSuggestions);
this.handleSuggestionsDisplayBehaviour();
```

I am not sure what the 2nd parameter for the `setFieldValue` does, here I am assuming you want to pass it false if there was no new suggestion.